src/main/java/com/cloudogu/scm/repositorytemplate/RepositoryTemplater.java

```diff
@@ -0,0 +1,187 @@
+    InputStream templatedContent = new ByteArrayInputStream(templateFilter.filter(content, filepath).toByteArray());
+    context.create(filepath).from(templatedContent);
+  }
+
+  private void copyFile(InputStream content, String filepath) throws IOException {
+    context.create(filepath).from(content);
+  }
+
+  private TemplateEngine setupTemplateEngine(RepositoryTemplate repositoryTemplate) {
+    String configEngine = repositoryTemplate.getEngine() != null ? repositoryTemplate.getEngine() : "mustache";
+    TemplateEngine engine = engineFactory.getEngine(configEngine);
+    if (engine == null) {
+      throw NotFoundException.notFound(ContextEntry.ContextBuilder.entity(TemplateEngine.class, configEngine));
+    }
+    return engine;
+  }
+
+  private RepositoryTemplate parseTemplateConfig(RepositoryService repositoryService) throws IOException {
+    Optional<String> templateFile = RepositoryTemplateFinder.templateFileExists(repositoryService);
+    if (!templateFile.isPresent()) {
+      throw NotFoundException.notFound(
+        ContextEntry.ContextBuilder.entity(RepositoryTemplate.class, "template file").in(repositoryService.getRepository())
+      );
+    }
+    try (InputStream templateYml = repositoryService.getCatCommand().getStream(templateFile.get())) {
+      RepositoryTemplate repositoryTemplate = unmarshallRepositoryTemplate(templateFile.get(), templateYml);
+      repositoryTemplate.setNamespaceAndName(repositoryService.getRepository().getNamespaceAndName().toString());
+      if (repositoryTemplate.getFiles() == null) {
+        repositoryTemplate.setFiles(Collections.emptyList());
+      }
+      return repositoryTemplate;
+    }
+  }
+
+  private RepositoryTemplate unmarshallRepositoryTemplate(String templateFile, InputStream templateYml) {
+    RepositoryTemplate repositoryTemplate = yaml.loadAs(templateYml, RepositoryTemplate.class);
+    if (repositoryTemplate == null) {
+      throw new YAMLException("repository template invalid -> could not parse " + templateFile);
+    }
+    return repositoryTemplate;
+  }
+
+  static class TemplateFilter {
+    private final InitializerContext context;
+    private final TemplateEngine engine;
+    private final Repository templateRepository;
+
+    private TemplateFilter(InitializerContext context, TemplateEngine engine, Repository templateRepository) {
+      this.context = context;
+      this.engine = engine;
+      this.templateRepository = templateRepository;
+    }
+
+    private ByteArrayOutputStream filter(InputStream inputStream, String filePath) {
+      try (Reader content = new InputStreamReader(inputStream)) {
+        Template template = engine.getTemplate(filePath, content);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        OutputStreamWriter writer = new OutputStreamWriter(baos, StandardCharsets.UTF_8);
+        template.execute(writer, context.getRepository());
```

Comment:
We should not use the repository as root model, because this makes it very hard to extend the model in the future. A better way is to use a wrapping object, which is easy to extend.

`{{repository.name}}` instead of `{{ name }}`